sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/apachecommons/collections/AbstractCollectionDecorator.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Portions Copyright (c) Microsoft Corporation
```

Comment:
we should track the source library version, for tracking future fixes, etc.

Where are we tracking the source version?

Reply:
I'll add source version in all common files, good catch!

Reply:
Discussed with Kushagra. [Updated](https://github.com/Azure/azure-sdk-for-java/pull/9168/commits/69e6342ce5ad07b65f3a4adb4d510182795f29ab) the cgmanifest with the version information, this file can be used as the central place to track version, instead of adding same info in all source files.